pulp_ansible/app/tasks/collections.py

```diff
@@ -1149,6 +1168,15 @@ def _pre_save(self, batch):
                 name = d_content.content.name
                 namespace, created = AnsibleNamespace.objects.get_or_create(name=name)
                 d_content.content.namespace = namespace
+                if d_content.d_artifacts:
+                    da = d_content.d_artifacts[0]
+                    # Check to see if avatar failed to download, update metadata if so
+                    if da.deferred_download:
+                        d_content.d_artifacts = None
+                        d_content.content.avatar_sha256 = None
+                    # Check to see if upstream didn't have avatar_sha256 set
+                    elif d_content.content.avatar_sha256 is None:
+                        d_content.content.avatar_sha256 = da.artifact.sha256
```

Comment:
The name of this class does not match anymore.

Reply:
It's called `CollectionContentSaver`, yet it manipulates all types of content that come flying by (not your fault, but we could take the chance and make it a `AnsibleContentSaver`.).